### ISSUE TITLE
Git Branches overlap when using .delete()

### ISSUE THREAD

**Issue by idantr** [nicoespeon/gitgraph.js#49]
Hi Guys,
We are using this opensource and it fits to our needs, it's great
but we have the issue of overlapping branches, in previous topics here you suggested using the .delete() once you do not use a branch any more,
I was using it after merging branches on the old branch that no longer continue...
but there is a bug once you .delete() a branch, in the next drawings if a commit has 2 children( 2 new branches that split from the master) the open source does not know to indent the second child anymore.

![image](https://cloud.githubusercontent.com/assets/10183179/5426847/d87a1430-837c-11e4-8404-478eae00f8dd.png)

in the example i used this code:

```
gitGraph.author = "Fabien0102 <fabien0102@planee.fr>";
master.commit();

/***********************
 *       MERGES        *
 ***********************/

master.checkout();

// Merge "dev" branch into HEAD (which is "master"), with a default message
dev.merge();

// Create a "test" branch and merge in into "master" with a custom message.
var test = gitGraph.branch( "test" );
test.commit( "Final commit" );
test.merge( master, "My special merge commit message" );
test.delete(); //This delete does not work as expected
var test2 = master.branch( "test2" );
test2.commit("test2");
var test3 = master.branch( "test3" );
test3.commit("test3");

// Then, continue committing on the "test" branch
test.commit( "It's works !" );

/***********************
 *       EVENTS        *
 ***********************/
```

this is how it looks without the delete()
![image](https://cloud.githubusercontent.com/assets/10183179/5426849/eaea641c-837c-11e4-935e-62bc6884f7f1.png)

i would be glad if you can suggest how to solve this,
 there are more cases like this but in general the delete does not work as expected
Regards
Idan Treivish

**Comment by sourtin** [comment]
I believe the reason for this is that, after a deletion, the calculColumn algorithm searches through all branches, deleted and undeleted, and returns the first column from a deleted branch. The problem is that it continues to use this column every time and so you get the behaviour seen in your first screenshot. Hopefully my [pull request](https://github.com/nicoespeon/gitgraph.js/pull/72) should fix this.

**Comment by ringlej** [comment]
@sourtin I solved the problem is a different way See my [pull request](https://github.com/nicoespeon/gitgraph.js/pull/71). My develop branch also adds some nice features to label branches and create tags

**Comment by sourtin** [comment]
Cool, you've got some very nice features in there :) I'm not sure that does solve the problem however: Consider someone creating four branches, `a`, `b`, `c` and `d`. Now delete branch `c` and create a new branch `e`, it will correctly go in the third column and you will again have four adjacent columns: `a b e d`. Now remove branches `d` and `e`. The `branchs` array is `[A(1) B(2) *(3) d(4) e(3)]` where capitals are live, asterisks are dead and lowercases are finished. If you now create a new branch `f` it will take the first finished column, 4, and so you will be left with a gap between branches `a b _ e`.

**Comment by ringlej** [comment]
Your right. I merged your patch into my develop branch. Thanks!

**Comment by sourtin** [comment]
No problem, thanks!

**Comment by nicoespeon** [comment]
Fixed with #72, thanks @sourtin :+1:
